eng/Tools.props

```diff
@@ -0,0 +1,19 @@
+<Project>
+
+  <!-- source-built packages -->
+  <ItemGroup>
+    <!-- arcade -->
+    <PackageReference Include="Microsoft.DotNet.Build.Tasks.Packaging" Version="$(MicrosoftDotNetBuildTasksPackagingVersion)" />
+  </ItemGroup>
+
+  <!-- excluded from source build -->
+  <ItemGroup Condition="'$(DotNetBuildFromSource)' != 'true'">
+    <PackageReference Include="Microsoft.DotNet.Build.Tasks.Feed" Version="$(MicrosoftDotNetBuildTasksFeedVersion)" />
+    <PackageReference Include="Microsoft.DotNet.VersionTools.Tasks" Version="$(MicrosoftDotNetVersionToolsTasksVersion)" />
+
+    <!-- SourceLink -->
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="$(MicrosoftSourceLinkVersion)" PrivateAssets="all" IsImplicitlyDefined="true" />
+    <PackageReference Include="Microsoft.SourceLink.AzureRepos.Git" Version="$(MicrosoftSourceLinkVersion)" PrivateAssets="all" IsImplicitlyDefined="true" />
+  </ItemGroup>
```

Comment:
Any idea why do we need this? Seems like something that should be handled by Arcade.

Reply:
@jmarolf I stole this from your PR. Is it necessary?

Reply:
hmm based on my run without the Tools.props ([rolsyn CI](https://dev.azure.com/dnceng/public/_build/results?buildId=1068681&view=logs&j=cec4b4e6-75b4-5c53-1ec7-3a9ace327284&t=c4465378-cdcd-58b8-bb7f-da7392fa5871)). I see Jon opened this issue https://github.com/dotnet/arcade/issues/7054

Reply:
I wonder if having this codintionally included is somehow the source of the issue https://github.com/dotnet/arcade/blob/0b4d6253f298a5f9d6e2c4bd20e1de33a6763061/src/Microsoft.DotNet.Arcade.Sdk/tools/Tools.proj#L32

Reply:
I am confused by this line https://github.com/dotnet/arcade/blob/0b4d6253f298a5f9d6e2c4bd20e1de33a6763061/src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj#L115. Is it removing the Publish property before the Tools.proj is restored?

Reply:
> I am confused by this line

I believe this is to override the default .NET SDK behaviors here and provide our own

Reply:
So boils down to Roslyn CI runs -restore separate from -publish. The Tools.proj is only Restored during a restore, but the "Microsoft.DotNet.Build.Tasks.Feed" package is conditioned on publish.